From 3b80baebc6e52a2feeafb5360c904ba01b155f5e Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Thu, 23 Oct 2014 10:12:41 -0400
Subject: [fs] ext4: no need to truncate pagecache twice in collapse range

Message-id: <1414059176-6044-18-git-send-email-lczerner@redhat.com>
Patchwork-id: 98290
O-Subject: [RHEL 7.1 PATCH 17/32] ext4: no need to truncate pagecache twice in collapse range
Bugzilla: 1150171

BZ 1150171
https://bugzilla.redhat.com/show_bug.cgi?id=1150171

BREW 8149772
https://brewweb.devel.redhat.com/taskinfo?taskID=8149772

Upstream commit 9337d5d31ab798f0c74150506371551a9195251a

 We're already calling truncate_pagecache() before we attempt to do any
 actual job so there is not need to truncate pagecache once more using
 truncate_setsize() after we're finished.

 Remove truncate_setsize() and replace it just with i_size_write() note
 that we're holding appropriate locks.

 Signed-off-by: Lukas Czerner <lczerner@redhat.com>
 Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/ext4/extents.c b/fs/ext4/extents.c
index 47bb147..ff313cc 100644
--- a/fs/ext4/extents.c
+++ b/fs/ext4/extents.c
@@ -5469,7 +5469,7 @@ int ext4_collapse_range(struct inode *inode, loff_t offset, loff_t len)
  }
 
  new_size = i_size_read(inode) - len;
- truncate_setsize(inode, new_size);
+ i_size_write(inode, new_size);
  EXT4_I(inode)->i_disksize = new_size;
 
  ext4_discard_preallocations(inode);
-- 
1.7.1